src/components/MovieForm/MovieForm.jsx

```diff
@@ -0,0 +1,255 @@
+import React, {useEffect, useState} from "react";
+import {Alert, Button, Form, Spinner} from "react-bootstrap";
+import {useParams} from "react-router-dom";
+import Page from "../Page";
+import {fileToBase64} from "../../utils/FileToBase64";
+import DatePicker from 'react-datepicker';
+import "react-datepicker/dist/react-datepicker.css";
+import "./MovieForm.css";
+import {backend} from "../../constants/backend";
+import {getHeaders} from "../../utils/CORSHeaders";
+
+const sendMovie = async (method, data, id) => {
+    let baseUrl = backend.MOVIES;
+    if (method === "PUT") {
+        baseUrl += `${id}/`;
+    }
+    const headers = getHeaders();
+    let res = await fetch(baseUrl, {
+        method: method,
+        headers,
+        body: JSON.stringify(data)
+    });
+    let status = method === "PUT" ? 200 : 201;
+    if (res.status === status) {
+        return await res.json();
+    } else {
+        throw res.status;
+    }
+};
+
+const getMovieDetails = async (id) => {
+    let baseUrl = backend.MOVIES + id + "/";
+    const headers = getHeaders();
+
+    let res = await fetch(baseUrl, {
+        method: "GET",
+        headers
+    });
+
+    if (res.status === 200) {
+        return await res.json();
+    } else {
+        throw res.status;
+    }
+};
+
+const mapResponseData = (source) => ({
+    name: source.movieName,
+    genre: source.movieGenre,
+    director: source.movieDirector,
+    description: source.movieDescription,
+    releaseDate: new Date( source.movieReleaseDate ),
+    image: source?.movieImage || null
```

Comment:
Nie wiem czy w końcu nie stanęło na tym, że jak nie ma zdjęcia to jest "" a nie null, ale tu nie mam pewności, trzeba by się było dopytać